integration-tests/ahm/src/tests.rs

```diff
@@ -832,3 +832,227 @@ fn test_account_references() {
+	});
+
+	// prepare the AH to send XCM messages to RC and Collectives.
+	ah.execute_with(|| {
+		ah_now = ah_now + 1;
+		frame_system::Pallet::<AhRuntime>::reset_events();
+		frame_system::Pallet::<AhRuntime>::set_block_number(ah_now);
+
+		// setup default XCM version
+		let result =
+			AhRuntimeCall::PolkadotXcm(pallet_xcm::Call::<AhRuntime>::force_default_xcm_version {
+				maybe_xcm_version: Some(xcm::prelude::XCM_VERSION),
+			})
+			.dispatch(AhRuntimeOrigin::root());
+		assert!(result.is_ok(), "fails with error: {:?}", result.err());
+	});
+
+	let (first_query_id, second_query_id, third_query_id) = rc.execute_with(|| {
+		assert_eq!(pallet_rc_migrator::PendingXcmMessages::<RcRuntime>::count(), 0);
+		// the query ids are incremented by 1 for each message.
+		(0, 1, 2)
+	});
+
+	// send invalid XCM message from RC to AH via rc-migrator.
+	let dmp_messages = rc.execute_with(|| {
+		rc_now = rc_now + 1;
+		frame_system::Pallet::<RcRuntime>::reset_events();
+		frame_system::Pallet::<RcRuntime>::set_block_number(rc_now);
+
+		let mut batch = pallet_rc_migrator::types::XcmBatchAndMeter::new_from_config::<RcRuntime>();
+		batch.push((None, vec![], vec![]));
+		// adding a second item; this will cause the dispatchable on AH to fail.
+		batch.push((None, vec![], vec![]));
+
+		pallet_rc_migrator::Pallet::<RcRuntime>::send_chunked_xcm_and_track(batch, |batch| {
+			pallet_rc_migrator::types::AhMigratorCall::<RcRuntime>::ReceiveReferendaValues {
+				values: batch,
+			}
+		})
+		.expect("failed to send XCM messages");
+
+		// make sure the message buffered in the rc migrator.
+		let message_hash = pallet_rc_migrator::PendingXcmQueries::<RcRuntime>::get(first_query_id)
+			.expect("query id not found");
+		assert!(pallet_rc_migrator::PendingXcmMessages::<RcRuntime>::get(message_hash).is_some());
+
+		// take the message from the queue to feed it later to the AH message processor.
+		let dmp_messages = DownwardMessageQueues::<RcRuntime>::take(AH_PARA_ID);
+		assert_eq!(dmp_messages.len(), 1);
+
+		dmp_messages
+	});
+
+	// process the message in the AH.
+	let ump_messages = ah.execute_with(|| {
+		ah_now = ah_now + 1;
+		frame_system::Pallet::<AhRuntime>::reset_events();
+		frame_system::Pallet::<AhRuntime>::set_block_number(ah_now);
+
+		enqueue_dmp((dmp_messages, 0u32.into()));
+
+		<asset_hub_polkadot_runtime::MessageQueue as OnInitialize<_>>::on_initialize(ah_now);
+		<asset_hub_polkadot_runtime::MessageQueue as OnFinalize<_>>::on_finalize(ah_now);
+
+		// take the acknowledgement message from the AH.
+		let ump_messages = PendingUpwardMessages::<AhRuntime>::take();
+		assert_eq!(ump_messages.len(), 1);
+
+		ump_messages
+	});
+
+	// process the acknowledgement message from AH in the RC.
+	rc.execute_with(|| {
+		rc_now = rc_now + 1;
+		frame_system::Pallet::<RcRuntime>::reset_events();
+		frame_system::Pallet::<RcRuntime>::set_block_number(rc_now);
+
+		enqueue_ump((ump_messages, 0u32.into()));
```

Comment:
Can we also check the content of this UMP?

Reply:
added event assertion